Server/Room.cpp

```diff
@@ -42,49 +52,108 @@ Room::Room(unsigned short id, Client networkInformations)
-
-    _worldInstance.get()->getEntity(entityId).addComponent<Networkable>(10);
-
-    Position &entityPosition = _worldInstance.get()->getEntity(entityId).getComponent<Position>();
-
-    entityPosition.modified = true;
-
-    _communicatorInstance.get()->startReceiverListening();
-    _worldInstance->addSystem<Temp>();
-    _worldInstance->addSystem<SendToClient>();
+    startConnexionProtocol();
+    initEcsGameData();
     _state = RoomState::LOBBY;
     while (_state != RoomState::ENDED && _state != RoomState::UNDEFINED) {
         try {
             connexionDemand = _communicatorInstance.get()->getLastMessage();
-            std::cerr << "Room " << _id << " received a connexion protocol."
-                      << std::endl; /// WILL BE DELETED WITH CONNEXION PROTOCOL ISSUE
-            _worldInstance->addEntity().addComponent<NetworkClient>(connexionDemand.message.clientInfo.getId());
-            _transisthorInstance.get()->transitEcsDataToNetworkDataEntityObstacle(entityId, entityPosition.x,
-                entityPosition.y, 5, std::string(""),
-                {connexionDemand.message.clientInfo.getId()}); /// USED FOR FUNCTIONNAL TESTING, WILL BE REMOVED LATER
+            if (connexionDemand.message.type == 10)
+                holdANewConnexionRequest(connexionDemand);
         } catch (NetworkError &error) {
         }
-        _worldInstance.get()->runSystems(); /// WILL BE IMPROVED IN PART TWO (THREAD + CLOCK)
+        if (_remainingPlaces != 4)
+            _worldInstance.get()->runSystems(); /// WILL BE IMPROVED IN PART TWO (THREAD + CLOCK)
+    }
+}
+
+void Room::holdANewConnexionRequest(CommunicatorMessage connexionDemand)
+{
+    if (_remainingPlaces == 0) {
+        _communicatorInstance.get()->sendDataToAClient(connexionDemand.message.clientInfo, nullptr, 0, 11);
+        return;
+    }
+    _remainingPlaces -= 1;
+    std::cerr << "Room " << _id << " received a connexion protocol." << std::endl;
+    std::size_t tempId = createNewPlayer(*_worldInstance.get(), 10, 10, 0, 0, 1, 5, 5, 45, 3, 4, "",
+        _worldInstance.get()
+            ->getResource<NetworkableIdGenerator>()
+            .generateNewNetworkableId()); /// CREATE A NEW ENTITY (INITIATED BY THE SERVER)
+    _worldInstance.get()->getEntity(tempId).addComponent<NetworkClient>(connexionDemand.message.clientInfo.getId());
+    std::vector<std::shared_ptr<Entity>> joined = _worldInstance.get()->joinEntities<Networkable>();
+
+    for (std::shared_ptr<Entity> entityPtr : joined) {
+        if (entityPtr->contains<AlliedProjectile>()) {
+            _worldInstance.get()->getTransisthorBridge()->transitEcsDataToNetworkDataEntityAlliedProjectile(
+                entityPtr->getComponent<Networkable>().id, entityPtr->getComponent<AlliedProjectile>().parentNetworkId, "",
+                {connexionDemand.message.clientInfo.getId()});
+        }
+        if (entityPtr->contains<Enemy>()) {
```

Comment:
In part 2, you may think about splitting this function for each if

Reply:
:eyes: